capsules/src/alarm.rs

```diff
@@ -147,6 +156,24 @@ impl<A: Alarm<'a>> Driver for AlarmDriver<'a, A> {
                         td.expiration = Expiration::Abs(time as u32);
                         (ReturnCode::SuccessWithValue { value: time }, true)
                     },
+                    5 /* Set relative expiration */ => {
+                        let mut time = now.wrapping_add (data as u32) as usize;
+                        let min_tics = self.min_tics.unwrap_or_else(|| {
+                            // scale the min tics from 16Khz to the actual frequency of the CPU
+                            let min_tics = (MIN_TICS_AT_16KHZ * (<A::Frequency>::frequency() as usize)) / 16000;
+                            self.min_tics.set(min_tics);
+                            min_tics
+                        });
+                        if time.wrapping_sub (now as usize) <= min_tics {
+                            time = time.wrapping_add (min_tics);
+                        }
```

Comment:
Note: In the future, we can simplify this part with the new time HIL (with the proposed `set_alarm(now, time)` function).

Reply:
I added a closure and used it in both of the commands.